scripts/wsrep_sst_clone.sh

```diff
@@ -458,6 +460,76 @@ setup_clone_plugin()
     wsrep_log_debug "-> ############## SSL SECTION [END] ($ROLE)############"
 }
 
+# Let's use the simple approach to detect if the transfer is running or not.
+# Monitor p_s.clone_progress table. If anything changes there, it means something
+# is happening.
+# One of the columns is DATA: The amount of data transferred in current state, in bytes,
+# so we should be good.
+# Allow transfer stuck for sst-idle-timeout
+monitor_sst_progress() {
+  local progress_monitor_command=$1
+  local timeout=$2
+  local clone_instance_pid=$3
+  local parent_process_pid=$4
+
+  local current_timeout=0
+  local previous_command_output=
+  local sleep=5
+  local previous_progress=
+
+  wsrep_log_debug "progress_monitor_command: ${progress_monitor_command}"
+  wsrep_log_debug "timeout: ${timeout}"
+  wsrep_log_debug "clone_instance_pid: ${clone_instance_pid}"
+  wsrep_log_debug "parent_process_pid: ${parent_process_pid}"
+
+  if [ ${timeout} -eq 0 ]; then
+    return 2;
+  fi
+
+  sleep=1
+  while true; do
+    # check if parent process is still alive
+    # Is it needed? If parent process dies, SST script dies as well
+    # and will cleanup clone instance in cleanup_joiner() trap.
+    kill -0 $parent_process_pid 2> /dev/null
+    if [ $? -ne 0 ]; then
+      return 1;
+    fi
+
+    # check if clone instance process is still alive
+    kill -0 $clone_instance_pid 2> /dev/null
+    if [ $? -ne 0 ]; then
+      # Clone instance is done with its job
+      return 0;
+    fi
+
+    # check if there is any SST progress
+    command_output=$(eval "$progress_monitor_command" 2> /dev/null ||:)
+    if [ "${command_output}" != "${previous_command_output}" ]; then
+        current_timeout=0
+    else
+        current_timeout=$((current_timeout + 1))
+        wsrep_log_debug "sst idle for ${current_timeout} sec"
+    fi
+
+    if [ ${current_timeout} -eq  ${timeout} ]; then
+      return 2
+    fi
+
+    # Report the current progress
+    current_progress=`$MYSQL_ACLIENT -NB -e "select format(((data/estimate)*100),2) 'completed%' from performance_schema.clone_progress where stage like 'FILE_COPY';" 2> /dev/null` || :
```

Comment:
mm not sure about this.
In very large datasets this % can be not good meaning that some changes may stay longer than 5 seconds.
like 20.45 can move to 20.46 in 10 seconds.
I would take the advantage of the fact we have many numbers after 0. and use them not formatting the %